### ISSUE TITLE
Combination Flying/Ground Mounts do not function properly

### ISSUE THREAD

**Issue by elliotmendiola** [dalaranwow/dalaran-wow#5735]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)

Mounts that function as both flying and ground mounts do not properly transition to flying when leaving a no fly zone, also during temporary no fly zones like Wintergrasp combination mounts fail to mount at all.

**Expected behaviour**: Tell us what should happen instead

Combination Mounts should automatically enter flying mode when leaving a no fly zone, and in temporary no fly zones they should be treated as ground mounts

**Steps to reproduce the problem**:

1. Use a combination mount like the Celestial Steed or Invincible while in Dalaran
2. Walk to Krasus landing and try to fly away, note you will have to dismount and remount to do so.

1. Use a combination mount during an active wintergrasp, note how you instantly dismount.

**Include proofs for this behaviour**

Would have to upload a video as it is based off an interaction that cannot be communicated in a still picture, steps to reproduce will reliably reproduce the behavior

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**
